app/create_file.py

```diff
@@ -1 +1,56 @@
-# write your code here
+from datetime import datetime
+from sys import argv
+from os import mkdir, chdir
+
+filename = ""
+directories = []
+
+if "-f" in argv:
+    index = 0
+
+    for _ in range(len(argv) - 1):
+        index += 1
+        if argv[_: _ + 1] == ["-f"]:
+            break
+
+    filename = argv[index]
+
+    if "-d" in argv:
+        index2 = 0
+
+        for _ in range(len(argv) - 1):
+            index2 += 1
+            if argv[_: _ + 1] == ["-d"]:
+                break
+
+        directories = argv[index2:index - 1]
+else:
+    filename = "file.txt"
+
+    if "-d" in argv:
+        index = 0
+
+        for _ in range(len(argv) - 1):
+            index += 1
+            if argv[_: _ + 1] == ["-d"]:
+                break
+
+        directories = argv[index:len(argv)]
+
+file_data = [datetime.today().strftime('%y:%d:%m %H-%M-%S'), "\n"]
+
+while True:
+    line = input("Enter content line: ")
+    if line == "exit":
+        break
+    file_data.extend([line])
+
+for directory in directories:
+    try:
+        mkdir(directory)
+    except Exception:
+        pass
+    chdir(directory)
```

Comment:
- Changing directories (chdir) multiple times is risky and can lead to unexpected results. It's better to construct the full path and create directories accordingly using `os.makedirs()` function which can take a relative path and create all directories in one go. For creating path from passed in directories use `os.path.join()` function passing all directory names.
To do that we can use `*` symbol which will unpack elements of the list and pass each of them as separate argument to function.

```suggestion
import os

dir_path = os.path.join(*directories)
if not os.path.exists(dir_path):
    os.makedirs(dir_path)
os.chdir(dir_path)
```

- A generic exception is caught, which is not good practice. Catch specific exceptions.
In that case if the directory already exists `FileExistsError` is raised. If a parent directory in the path does not exist, `FileNotFoundError` is raised.